### PR TITLE
TRUNK-5848: Remove raw use of parameterized class

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
## Description of what I changed
Remove raw use of parameterized classes: List, LinkedHashSet, HashSet, Map, Set, Map.Entry

## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-5848

## Checklist: I completed these to help reviewers :)
- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
@BartlomiejRasztabiga i feel like you have done enough intro tickets to move on to more complicated ones. For instance, you can get started on low complexity tickets but that do not have the `intro` label.

**Comment by BartlomiejRasztabiga** [comment]
@dkayiwa Understood, I will try to do some work on more complicated tasks for sure :)

**Comment by dkayiwa** [comment]
@BartlomiejRasztabiga that is when you will learn more. 😊

File: api/src/main/java/org/openmrs/notification/impl/MessageServiceImpl.java [hunk 1]
```diff
@@ -274,7 +274,7 @@ public Message prepareMessage(String templateName, Map data) throws MessageExcep
 	 */
 	@Override
 	@Transactional(readOnly = true)
-	public List getAllTemplates() throws MessageException {
+	public List<Template> getAllTemplates() throws MessageException {
```
Comment (sacull) [inline api/src/main/java/org/openmrs/notification/impl/MessageServiceImpl.java:277]: In your free time, take a look here: https://github.com/openmrs/openmrs-core/pull/3368#pullrequestreview-451799315 ;)

Comment (BartlomiejRasztabiga) [reply api/src/main/java/org/openmrs/notification/impl/MessageServiceImpl.java]: Got it, I've just reverted changes for MessageService and MessageServiceImpl

**Comment by gracepotma** [comment]
@sacull would you mind reviewing this PR since it's been updated since you last provided feedback? Would be nice to close this :)

**Pull request merged**